src/pudl/transform/nrelatb.py

```diff
@@ -0,0 +1,417 @@
+"""Transform NREL ATB data into well normalized, cleaned tables."""
+
+from typing import Self
+
+import pandas as pd
+from dagster import asset
+from pydantic import BaseModel, field_validator
+
+import pudl.helpers as helpers
+
+IDX_ALL = [
+    "report_year",
+    "core_metric_case",
+    "core_metric_variable_year",
+    "cost_recovery_period_years",
+    "scenario_atb",
+    "technology_description",
+    "technology_description_detail_1",
+    "technology_description_detail_2",
+    "core_metric_parameter",
+]
+"""Expected primary key columns for the raw nrelatb data.
+
+The normalized core tables we are trying to build will have primayr keys which
+are a subset of these columns.
+"""
+
+
+class TableNormalizer(BaseModel):
+    """Info needed to convert a selection of the raw NREL table into a normalized table."""
+
+    idx: list[str]
+    """Primary key columns of normalized subset table."""
+    columns: list[str]
+    """Columns reported in raw NREL table that are unique to the :attr:`idx`."""
+
+
+def transform_normalize(nrelatb: pd.DataFrame, normalizer: TableNormalizer):
+    """Normalize a subset of the NREL ATB data into a small table.
+
+    Given a :class:`TableNormalizer` with a set of primary keys (``idx``) and a
+    list of columns (``columns``), build a table with just those primary key and
+    data columns from the larger ATB semi-processed table (output of
+    :func:`_core_nrelatb__transform_start`). Ensure that the output table is
+    unquie absed on the primary keys.
```

Comment:
```suggestion
    unique based on the primary keys.
```
I just installed [this spell checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker). Seems like there's a bunch of false positives for things like "zenodo," but it's relatively easy to add stuff to the user dictionary 🤷 idk, might be useful?

Reply:
omigosh i've needed this for so long great suggestion i cannot spell v well